backend/src/main/java/ai/giskard/web/socket/WorkerStatusSocketService.java

```diff
@@ -18,19 +17,35 @@ public class WorkerStatusSocketService {
     private final MLWorkerService mlWorkerService;
     private final SimpMessagingTemplate simpMessagingTemplate;
 
-    @PostConstruct
-    public void init() {
-        Timer timer = new Timer();
-        timer.schedule(new TimerTask() {
-            @Override
-            public void run() {
-                Map<String, Boolean> data = new HashMap<>();
-                data.put("connected", mlWorkerService.isExternalWorkerConnected());
-
-                simpMessagingTemplate.convertAndSend(
-                        "/topic/worker-status",
-                        data);
-            }
-        }, 0, 5000);
+    @EventListener
+    public void handleSessionSubscribeEvent(SessionSubscribeEvent event) {
+        sendCurrentStatus();
+    }
+
+    @EventListener(condition = "#event.isConnected == true")
+    public void handleWorkerConnectedEvent(UpdateWorkerStatusEvent event) {
+        Map<String, Boolean> data = new HashMap<>();
+        data.put("connected", true);
+        sendData(data);
+    }
+
+    @EventListener(condition = "#event.isConnected == false")
+    public void handleWorkerDisconnectedEvent(UpdateWorkerStatusEvent event) {
+        Map<String, Boolean> data = new HashMap<>();
+        data.put("connected", false);
+        sendData(data);
+    }
```

Comment:
We can simply get the status from the event and one listener to forward it (we will need to add a getter in the event)

```suggestion
    @EventListener()
    public void handleWorkerStatusChangeEvent(UpdateWorkerStatusEvent event) {
        Map<String, Boolean> data = new HashMap<>();
        data.put("connected", event.isConnected());
        sendData(data);
    }
```